pickly-service/src/main/java/org/pickly/service/friend/service/impl/FriendServiceImpl.java

```diff
@@ -40,4 +40,25 @@ private void checkAlreadyFriend(final Long followerId, final Long memberId) {
     }
   }
 
+  private Friend checkIsFollower(final Long followerId) {
+    Friend friend = friendRepository.findById(followerId)
+        .orElseThrow(() -> new BusinessException("팔로우 중이 아닙니다", ErrorCode.INVALID_INPUT_VALUE));
+    return friend;
+  }
+
+  @Override
+  @Transactional
+  public void enableNotification(Long followerId, Long memberId) {
+    Friend follower = checkIsFollower(followerId);
+
+    follower.enableNotification();
+  }
+
+  @Override
+  @Transactional
+  public void disableNotification(Long followerId, Long memberId) {
+    Friend follower = checkIsFollower(followerId);
+
+    follower.disableNotification();
+  }
```

Comment:
다음과 같은 로직도 고려해볼 수 있을 것 같아요! member에서 friend를 주입받아 해당 friend랑 상호작용하는 방향으로 코드를 작성할 수 있을 것 같습니다
서비스는 가능한 도메인 로직에 데이터를 받아서 넣어주는 역할을 하고 도메인에서 로직을 수행할 수 있도록 만들게요
```suggestion
  @Override
  @Transactional
  public void switchNotificationAllowStatus(Long followerId, Long memberId) {
    Member member = memberRepository.findById(memberId);
    Friend friend = friendReository.findById(followerId);

    member.switchNotificationAllowSatus(friend);
  }
```
자세한 내용은 다음 영상을 보시면 도움이 될 것 같습니다
https://www.youtube.com/watch?v=26S4VFUWlJM

Reply:
오 참고해서 수정해보겠습니다 🙇‍♂️

Reply:
서경님 일단 오늘 리팩토링을 해보려고 했는데 이 말씀이 혹시 Member로 Friend를 가져가면 좋겠다고 이해하면 될까요?
이해가 잘안가서 시간 나실 때 설명 한 번만 부탁드려도 될까요? 😅

Reply:
제 말은 로직이 서비스가 아니라 도메인단으로 올렸으면 좋겠다고 말씀 드린 거에요 member를 넣은 건 그냥 memberId가 있길래 이걸로 Member를 찾으려는 건가? 생각한 거고요 ㅋㅋ